synapse/app/_base.py

```diff
@@ -124,6 +124,19 @@ def quit_with_error(error_string):
     sys.exit(1)
 
 
+def listen_metrics(bind_addresses, port):
+    """
+    Start Prometheus metrics server.
+    """
+    from synapse.metrics import RegistryProxy
+    from prometheus_client import start_http_server
+
+    for host in bind_addresses:
+        reactor.callInThread(start_http_server, int(port),
+                             addr=host, registry=RegistryProxy)
```

Comment:
`start_http_server` should spawn its own thread, so is it necessary to `callInThread`?

Reply:
I had issues with it jamming Twisted, for some reason, and this made it work fine. ~the mysteries of threading~